lib/finch/http1/conn.ex

```diff
@@ -174,33 +175,83 @@ defmodule Finch.Conn do
         Telemetry.stop(:recv, start_time, metadata, extra_measurements)
         {:ok, %{conn | mint: mint}, acc}
 
+      {:error, _mint, error = :response_body_too_large} ->
+        metadata = Map.put(metadata, :error, error)
+        Telemetry.stop(:recv, start_time, metadata, extra_measurements)
+        {:error, close(conn), Error.exception(error)}
+
       {:error, mint, error} ->
         metadata = Map.put(metadata, :error, error)
         Telemetry.stop(:recv, start_time, metadata, extra_measurements)
         {:error, %{conn | mint: mint}, error}
     end
   end
 
-  defp receive_response([], acc, fun, mint, ref, timeout) do
+  defp receive_response(entries, acc, fun, mint, ref, timeout, max_body, body_sum \\ 0)
+
+  defp receive_response([], acc, fun, mint, ref, timeout, max_body, body_sum) do
     case HTTP1.recv(mint, 0, timeout) do
```

Comment:
The `0` passed in here is the number of bytes you want to accept from the socket. I believe a more correct implementation would pass a value here. See https://hexdocs.pm/mint/Mint.HTTP.html#recv/3

Reply:
You could set an upper bound. Some hints are that the maximum binary size returned could match the kernel receive buffer size. Maybe `65_536` would be good enough since the overhead of a TCP or SSL socket would be similar in size.